stattest/experiment/pipeline.py

```diff
@@ -0,0 +1,42 @@
+from multiprocessing import Event, Manager, Process, Queue
+
+from tqdm import tqdm
+
+
+def __show_prog(queue: Queue, shutdown_event, total):
+    prog = tqdm(total=total, desc="Processing data", unit_scale=True)
+    while not (shutdown_event.is_set() and queue.empty()):
+        try:
+            to_add = queue.get(timeout=1)
+            prog.update(to_add)
+            if prog.n >= total:
+                break
+        finally:
+            pass
+
+
+def start_pipeline(fill_queue, process_entries, num_workers, **kwargs):
+    queue_manager = Manager()
+    queue = queue_manager.Queue(maxsize=2000)
+    info_queue = queue_manager.Queue(maxsize=2000)
+    shutdown_event = Event()
```

Comment:
Is the ```shutdown_event``` event really needed? It is used only in ```process_entries``` function as one of the "stoppers" of performing useful work, but it is already set when the function is called, so it seems useless. ```fill_queue``` function sets it, but it is called before multiple processes are generated, so there is no collision. Or is it a reserve for the future?

Reply:
This trick will be useful when queue will be filled in another process